### ISSUE TITLE
[BUG] Amethyst ignores TOR settings and routes traffic directly

### ISSUE THREAD

**Issue by Perlover** [vitorpamplona/amethyst#1240]
**Describe the bug**  
In the latest version of Amethyst, the app does not respect TOR settings for an account configured to route traffic through an external TOR (port 9050). Even when the Orbot application (the external TOR service) is not running, Amethyst still loads posts, images, and notes for the account.  

**To Reproduce**  
1. Configure multiple accounts in Amethyst.  
2. Set one account to use an external TOR service via port 9050.  
3. Ensure the Orbot application is not running.  
4. Switch to the account configured to use TOR and observe that posts, images, and notes still load.  

**Expected behavior**  
Amethyst should not load any content for an account set to route traffic through TOR if the external TOR service is unavailable.  

**Video and Screenshots**  
Unfortunately, no screenshots or videos are available for this issue. Let me know if logs are needed for debugging.  

**Device (please complete the following information):**  
- App Version: 0.93.1

**Bounty (in Bitcoin sats) offered for a solution**  
No bounty offered at the moment, but this issue is important as it directly affects privacy and security.

**Comment by vitorpamplona** [comment]
The way the Tor library works is that if Orbot isn't running, the internal Tor will start at the same port the external Tor was set up. There doesn't seem to be a way to change it. So, It is impossible for us to know if the Proxy replying from 9050 is an external or internal. This means that if you selected External config and for some reason closed Tor, the internal Tor will take over. But the connection will never be clear net. 

Last time I saw, Orbot did provide a VPN service that blocked some apps from connecting. Are those things still available? If so, that is the way to block apps from accessing the web if Orbot is disabled. There are other apps, like https://rethinkdns.com/, that allow you to setup a similar blocker.

**Comment by Perlover** [comment]
Good afternoon. I also noticed that Orbot recently switched to VPN mode and started routing all traffic from the Android device through Tor. Therefore, at some point, I changed its settings and configured it to work exclusively as a proxy, not as a VPN, responding on port 9050.

However, when I was testing Amethyst, Orbot was not running at all, and I didn’t check how it behaves with Orbot enabled. To clarify, I did test it by launching Orbot, but since the latest version of Orbot does not display statistics for traffic sent and received through the proxy, I couldn’t determine whether Amethyst was routing traffic through Orbot or not.

Now that you’ve mentioned this behavior as a feature of the Tor library within Amethyst, it might explain everything. Still, I think this is not the best user experience. When Amethyst settings explicitly state that it should use port 9050, and nothing is running there, if the user sees traffic being successfully sent and received, many will likely suspect that their traffic is not protected by Tor.

Additionally, it seems impossible to verify whether the traffic is routed through Tor or not. This, frankly, makes it hard to be certain, for example, whether the traffic is going through Tor at all.

In any case, thank you for your response. The ticket can probably be closed if you are certain that when Orbot is not running, traffic from Amethyst is routed through the internal Tor library.

**Comment by vitorpamplona** [comment]
The only way to truly verify if an app is using Tor or not is by using monitoring apps like rethinkdns.com. 

We might need to make the settings page more clear though.

**Comment by Perlover** [comment]
> Please @Perlover can you check if you still have this issue in the latest version of Amethyst?

I just updated Amethyst to version 0.94.3 from the app store. After that, I launched it. First, I opened the accounts that don’t use TOR, and everything works fine there. Then I opened the account that I have protected with TOR, where the settings are configured to use Orbot via port 9050. However, Orbot is currently not running on my device.

When I scroll through the feed, it still loads. The latest posts from the people I follow are being fetched, and images are loading as well. So, it seems like everything is working just as it did before.

I haven’t tried the utility that Victor recommended earlier. At the same time, although Orbot is installed on my Android phone, it’s not currently running. Therefore, either everything is still functioning through the internal TOR library, as Victor mentioned, or there is still a potential leak.

It seems to me that if I have Orbot configured in the settings with port 9050, then it should work that way and not use the internal TOR.

**Comment by Perlover** [comment]
I am still testing the Amethyst app, version 0.94.3. I installed the utility recommended by Victor, called RethinkDNS, through FDroid.

If I use the option for the account "Use external Orbot" with port 9050, the utility shows that the traffic either goes through only two IP addresses, or in some cases, it doesn’t show any IP addresses at all. However, if I switch to an account where I force the use of the internal TOR with full privacy settings enabled, the RethinkDNS utility doesn’t show any IP addresses where the traffic is going.

In other words, it seems that both the internal TOR and the external Orbot configuration with port 9050 are working as expected. But if Orbot is not running, it looks like the internal TOR library is being used instead.

**Comment by KotlinGeekDev** [comment]
> I am still testing the Amethyst app, version 0.94.3. I installed the utility recommended by Victor, called RethinkDNS, through FDroid.
> 
> If I use the option for the account "Use external Orbot" with port 9050, the utility shows that the traffic either goes through only two IP addresses, or in some cases, it doesn’t show any IP addresses at all. However, if I switch to an account where I force the use of the internal TOR with full privacy settings enabled, the RethinkDNS utility doesn’t show any IP addresses where the traffic is going.
> 
> In other words, it seems that both the internal TOR and the external Orbot configuration with port 9050 are working as expected. But if Orbot is not running, it looks like the internal TOR library is being used instead.

Nice. So everything is working as expected.

**Comment by davotoula** [comment]
@Perlover happy to close this?

**Comment by Perlover** [comment]
Sorry, I am closing ticket!
